tasks/list-remaining-migrated-removals.ts

```diff
@@ -0,0 +1,207 @@
+          const balance = await removalContract.balanceOf(
+            signerAddress,
+            tokenId
+          );
+          return { tokenId, balance };
+        })
+      );
+      // filter out token ids that have a balance of 0
+      const listableData = remainingBalanceData.filter((data) =>
+        data.balance.gt(0)
+      );
+
+      const listableTokenIds = listableData.map((data) => data.tokenId);
+      const listableBalances = listableData.map((data) => data.balance);
+      // sum the listable balances and convert to ether
+      const totalListableBalance = listableBalances.reduce(
+        (accumulator, balance) => accumulator.add(balance),
+        ethers.BigNumber.from(0)
+      );
+      const totalListableBalanceInEther =
+        ethers.utils.formatEther(totalListableBalance);
+
+      hre.log(
+        chalk.white(
+          `🔎 Found ${listableTokenIds.length} removal tokens with a total listable balance of ${totalListableBalanceInEther} NRTs`
+        )
+      );
+      // if there are no listable token ids, exit
+      if (listableTokenIds.length === 0) {
+        hre.log(
+          chalk.white(
+            `👋 No listable token ids found (no non-zero balances), exiting without listing any removals...`
+          )
+        );
+        return;
+      }
+      if (!dryRun) {
+        hre.log(chalk.white(`🤞 Submitting multicall consign transaction...`));
+        let txResult;
+        let pendingTx: ethers.ContractTransaction;
+        try {
+          pendingTx = await removalContract.multicall(
+            listableTokenIds.map((id, index) =>
+              removalContract.interface.encodeFunctionData('consign', [
+                signerAddress,
+                id,
+                listableBalances[index],
+              ])
+            )
+          );
+          hre.log(`txHash: ${chalk.green(pendingTx.hash)}`);
+          hre.log(chalk.white('\n👷 Waiting for transaction to finalize...'));
+
+          const result = await pendingTx.wait(); // TODO specify more than one confirmation?
+          const txReceipt =
+            await removalContract.provider.getTransactionReceipt(
+              result.transactionHash
+            );
+          // if the status is 1, log a success message, otherwise log a failure
+          if (txReceipt.status === 1) {
+            hre.log(
+              chalk.green(
+                `✅ Successfully listed ${totalListableBalanceInEther} NRTs across ${listableTokenIds.length} removals!`
+              )
+            );
+          } else {
+            hre.log(
+              chalk.red(
+                `❌ Failed to list the removals! Check out the transaction receipt in the output.`
+              )
+            );
+          }
+          txResult = txReceipt;
+        } catch (error) {
+          hre.log(
+            chalk.red('❌ Error submitting multicall consign transaction: '),
+            error
+          );
+          txResult = error;
```

Comment:
should this throw?

Reply:
yeah - forgot to bring this script in line with the others.  it'll now write some information to a file and then crash and throw